fix(functions): 🐛 stalker prompt shows and no multiple stalkers

gchamb opened the pull request:
This shows the stalker's prompt answer and fixes the multiple stalker/spectator problem.

ignore the branch name should just be gchamb/fix-stalker

gchamb committed:
fix: :bug: show stalker prompt answer and fix multiple stalker bug

gchamb committed:
fix: prettier fix

mrjoshua520 reviewed (dismissed):
Looks great! Someone else should look at it before merging though

nstringham reviewed (dismissed):
we decided we should not delete prompt answers at the end pf prompt and use that at the beginning of vote and we should also delete chat messages asyncrounstly after setting the state t vote

gchamb committed:
refactor(functions): ♻️ async delete prompt answers

gchamb committed:
refactor(functions): just rearranged order

gchamb committed:
refactor(functions): ♻️ async delete chat collections

nstringham commented on functions/src/lobby.ts:
```diff
@@ -510,7 +523,13 @@ export const verifyExpiration = functions.https.onCall(async (data, context): Pr
     }
   });
+
+  if (lobby?.state == "CHAT") {
+    await deleteChatCollections(lobbyDocRef);
+  }
+
   if (lobby?.state == "VOTE") {
+    await deletePromptAnswers(lobbyDocRef);
```
why are prompt answers being deleted twice?

nstringham commented on functions/src/chat.ts:
```diff
@@ -19,44 +20,50 @@ export async function deleteLobbyChatMessages(lobbyDoc: DocumentReference<Lobby>
+  for (const chatRoomDoc of chatRooms.docs) {
+    // place messages in batch
+    const chatMessages = await getChatRoomMessagesCollection(chatRoomDoc.ref).get();
+    chatMessages.docs.map((messageDoc) => batch.delete(messageDoc.ref));
```
this does not return anything so it should be a map.

maybe conside a for of loop

nstringham commented on functions/src/chat.ts:
```diff
@@ -19,44 +20,50 @@ export async function deleteLobbyChatMessages(lobbyDoc: DocumentReference<Lobby>
 
-  for (const stalker of stalkers.docs) {
-    transaction.update(stalker.ref, { stalker: false });
+    if (promptAns === "" || !promptAnswerDoc.exists) {
```
don't need to check `!promptAnswerDoc.exists` because querries already garentee that the documents exist

nstringham commented on functions/src/chat.ts:
```diff
@@ -19,44 +20,50 @@ export async function deleteLobbyChatMessages(lobbyDoc: DocumentReference<Lobby>
+    }
+
+    const playerIndex = uids.indexOf(promptAnswerDoc.id);
+    players[playerIndex].promptAnswer = promptAns;
```
I feel like this could just be `players[playerIndex].promptAnswer = promptData.answer ?? "no answer"`

nstringham reviewed (dismissed):
one disavantage of what we are doing here iswe read the prompt awnsers twice at the end of chat once in `setAnswers` and once in `deletePromptAnswers` we could store if and reuse it but I think this is probobly fine for now

gchamb committed:
refactor(functions): ♻️ minor fixes for prompt answers

- fixed where prompt answres are only read once in setAndDeleteAnswers function

- Deletes the player's prompt answer doc after it is read

- reformat map into for of loop

- removes unneeded if statement

gchamb committed:
fix(functions): ♻️ delete the player's promptanswer from the previous round

gchamb committed:
refactor: prettier fix

gchamb committed:
fix(functions): 🐛 fix typo
